### PR TITLE
PSAP-424: Add driver-toolkit to OpenShift docs

### PR DESCRIPTION
**Pull request by dagrayvid**
As a part of [PSAP-424](https://issues.redhat.com/browse/PSAP-424)
https://issues.redhat.com/browse/OSDOCS-2237

This PR contains a first draft of documentation for the driver-toolkit image. As discussed on slack with @ahardin-rh we likely want to move the driver-toolkit and NFD docs out of performance and scalability into to a new "hardware enablement" section.

@ArangoGutierrez FYI

### CODE REVIEW

File: modules/psap-driver-toolkit-using.adoc
Comment (ashishkamra) [inline modules/psap-driver-toolkit-using.adoc:29]: I understand this is an example but still will be a good idea to follow Dockerfile best practices (unless that does not really work) - 1) single RUN command and chain individual commands using && 2) perform cleanup (remove yum cache, yum clean all etc)

File: modules/psap-driver-toolkit-pulling.adoc
Comment (ashishkamra) [inline modules/psap-driver-toolkit-pulling.adoc:11]: There are instructions at - https://catalog.redhat.com/software/containers/openshift4/driver-toolkit-rhel8/604009d6122bd89307e00865?tag=v4.7.0-202104142050.p0&push_date=1619452122000&container-tabs=gti - on how to pull this image. Should we just refer to that ? Which one is the source of truth ?

**Review by ashishkamra — CHANGES_REQUESTED** [review]

File: modules/psap-driver-toolkit-pulling.adoc
Comment (dagrayvid) [reply modules/psap-driver-toolkit-pulling.adoc:11]: Thanks for pointing that out, I think these instructions are sufficient and we should refer to them.

File: modules/psap-driver-toolkit-using.adoc
Comment (dagrayvid) [reply modules/psap-driver-toolkit-using.adoc:29]: Good point, I was able to combine several run commands and clear the dnf cache for the dnf installs.

File: modules/psap-driver-toolkit-pulling.adoc
Comment (ahardin-rh) [inline modules/psap-driver-toolkit-pulling.adoc:8]: available from the link:https://registry.redhat.io/[Container images section of the Red Hat Ecosystem Catalog]

the version number in the catalog.

File: modules/psap-driver-toolkit-pulling.adoc
Comment (ahardin-rh) [inline modules/psap-driver-toolkit-pulling.adoc:27]: [source,terminal]
    ----
    $ oc adm release info 4.8.0 --image-for=driver-toolkit
    ----
    +
     .Example output
    [source,terminal]
    ----
    quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:&lt;SHA&gt;
    -----

File: modules/psap-driver-toolkit-pulling.adoc
Comment (ahardin-rh) [inline modules/psap-driver-toolkit-pulling.adoc:31]: , such as the pull secret required

File: modules/psap-driver-toolkit-using.adoc
Comment (ahardin-rh) [inline modules/psap-driver-toolkit-using.adoc:8]: We can't link to GitHub repos without approval from Support. @sferich888  PTAL

File: modules/psap-driver-toolkit-using.adoc
Comment (ahardin-rh) [inline modules/psap-driver-toolkit-using.adoc:21]: . Create a namespace. For example:
    +
    [source,terminal]-
    ----
    $ oc new-project simple-kmod-demo
    -----

File: modules/psap-driver-toolkit-using.adoc
Comment (ahardin-rh) [inline modules/psap-driver-toolkit-using.adoc:27]: YAML file defines

File: modules/psap-driver-toolkit-using.adoc
Comment (ahardin-rh) [inline modules/psap-driver-toolkit-using.adoc:102]: . Substitute the correct


s/Openshift/{product-title} version

File: modules/psap-driver-toolkit-using.adoc
Comment (ahardin-rh) [inline modules/psap-driver-toolkit-using.adoc:106]: Each of these commands needs to be in a separate code block per https://github.com/openshift/openshift-docs/blob/master/contributing_to_docs/doc_guidelines.adoc#code-blocks-command-syntax-and-example-output

File: modules/psap-driver-toolkit-using.adoc
Comment (ahardin-rh) [inline modules/psap-driver-toolkit-using.adoc:116]: This should be a numbered step as well, I think

s/ImageStream/image stream
s/BuildConfig/build config
per https://github.com/openshift/openshift-docs/blob/master/contributing_to_docs/doc_guidelines.adoc#general-references

File: modules/psap-driver-toolkit-using.adoc
Comment (ahardin-rh) [inline modules/psap-driver-toolkit-using.adoc:122]: s/we can deploy/deploy
s/needs to run/must run
s/.yaml file/ YAML file

File: modules/psap-driver-toolkit-using.adoc
Comment (ahardin-rh) [inline modules/psap-driver-toolkit-using.adoc:189]: This can be a numbered step too

File: modules/psap-driver-toolkit-using.adoc
Comment (ahardin-rh) [inline modules/psap-driver-toolkit-using.adoc:195]: s/we can verify/verify

File: modules/psap-driver-toolkit-using.adoc
Comment (ahardin-rh) [inline modules/psap-driver-toolkit-using.adoc:196]: Break these up into separate commands and output blocks per https://github.com/openshift/openshift-docs/blob/master/contributing_to_docs/doc_guidelines.adoc#code-blocks-command-syntax-and-example-output

File: modules/psap-driver-toolkit.adoc
Comment (ahardin-rh) [inline modules/psap-driver-toolkit.adoc:10]: payload used as a base image on which you can build

Comment (ahardin-rh) [reply modules/psap-driver-toolkit.adoc:10]: modules, as well as

Comment (ahardin-rh) [reply modules/psap-driver-toolkit.adoc:10]: s/RHCOS nodes/{op-system-first} nodes

File: modules/psap-driver-toolkit.adoc
Comment (ahardin-rh) [inline modules/psap-driver-toolkit.adoc:12]: s/OCSs/operating systems like {op-system}

Comment (ahardin-rh) [reply modules/psap-driver-toolkit.adoc:12]: software-defined storage (SDS) solutions, such as Lustre parallel file systems, which require kernel modules on client machines.

Comment (ahardin-rh) [inline modules/psap-driver-toolkit.adoc:12]: What does FPGA stand for?

File: modules/psap-driver-toolkit.adoc
Comment (ahardin-rh) [inline modules/psap-driver-toolkit.adoc:16]: package names should be in back-ticks per https://github.com/openshift/openshift-docs/blob/master/contributing_to_docs/doc_guidelines.adoc#quick-markup-reference

File: modules/psap-driver-toolkit.adoc
Comment (ahardin-rh) [inline modules/psap-driver-toolkit.adoc:29]: Driver Toolkit

File: modules/psap-driver-toolkit.adoc
Comment (ahardin-rh) [inline modules/psap-driver-toolkit.adoc:39]: Prior to the Driver Toolkit's existence, you could install kernel packages in a pod or build config on {product-title} using

Comment (ahardin-rh) [reply modules/psap-driver-toolkit.adoc:39]: host's `machine-os-content` package  (is it a package?)

s/allows us to avoid/avoids 

s/Pod/pod

Comment (ahardin-rh) [reply modules/psap-driver-toolkit.adoc:39]: s/OpenShift/{product-title}

File: modules/psap-driver-toolkit.adoc
Comment (ahardin-rh) [inline modules/psap-driver-toolkit.adoc:41]: s/operator/Operator

Comment (ahardin-rh) [reply modules/psap-driver-toolkit.adoc:41]: s/plugin/plug-in
s/BuildConfig/build config

Comment (ahardin-rh) [reply modules/psap-driver-toolkit.adoc:41]: create _recipies_, or recommendations, for building

File: modules/psap-driver-toolkit-pulling.adoc
Comment (ahardin-rh) [inline modules/psap-driver-toolkit-pulling.adoc:24]: In general, procedures must be presented as a series of numbered steps, so we should reformat each procedure in that way

**Comment by ahardin-rh** [comment]
Is there a JIRA card tracking this work? If so, we should add the link to the top of the PR.

Just some miscellaneous style comments from me. Thanks for this!

**Comment by dagrayvid** [comment]
Thanks @ahardin-rh for the fixes! I think they should all be addressed. I also organized some procedure steps into substeps in `modules/psap-driver-toolkit-using.adoc`. 

With regards to the reference to the simple-kmod github repository, if this is not approved, we could remove the example usage section and save that for a blog post.

File: modules/psap-driver-toolkit.adoc
Comment (ahardin-rh) [reply modules/psap-driver-toolkit.adoc:46]: @dagrayvid Just spoke with Eric and we're going to have to remove the link. Thanks!

Comment (sferich888) [reply modules/psap-driver-toolkit.adoc:46]: I am not a fan of this link; because the link has no context on how to use it or what it solves; I don't think including thi is a good next step.

Comment (dagrayvid) [reply modules/psap-driver-toolkit.adoc]: Makes sense! removed.

File: modules/psap-driver-toolkit.adoc
Comment (ahardin-rh) [inline modules/psap-driver-toolkit.adoc:41]: This GitHub link must also be removed

File: modules/psap-driver-toolkit-pulling.adoc
Comment (ahardin-rh) [inline modules/psap-driver-toolkit-pulling.adoc:8]: ` driver-toolkit` image

File: modules/psap-driver-toolkit-pulling.adoc
Comment (ahardin-rh) [inline modules/psap-driver-toolkit-pulling.adoc:13]: `driver-toolkit` image

File: modules/psap-driver-toolkit-using.adoc
Comment (ahardin-rh) [reply modules/psap-driver-toolkit-using.adoc:8]: Confirmed that we must remove this GH link too

File: modules/psap-driver-toolkit-using.adoc
Comment (ahardin-rh) [inline modules/psap-driver-toolkit-using.adoc:27]: ` simple-kmod` driver

File: modules/psap-driver-toolkit-using.adoc
Comment (ahardin-rh) [inline modules/psap-driver-toolkit-using.adoc:124]: and build config:

File: modules/psap-driver-toolkit-using.adoc
Comment (ahardin-rh) [inline modules/psap-driver-toolkit-using.adoc:226]: s/Then execute the lsmod command/Execute the `lsmod` command

**Comment by ahardin-rh** [comment]
@dagrayvid Thanks for the updates! This is looking good. I have a few minor follow-up comments. Once all of those edits are complete and we get QE review, we should be set. We'll also need to squash the multiple commits down into one before we can merge. Thanks!

@wabouhamad Can you please provide a QE review?

**Comment by dagrayvid** [comment]
> @dagrayvid Thanks for the updates! This is looking good. I have a few minor follow-up comments. Once all of those edits are complete and we get QE review, we should be set. We'll also need to squash the multiple commits down into one before we can merge. Thanks!
> 
> @wabouhamad Can you please provide a QE review?

Fixed the comments and squashed the commits, thanks!

**Comment by dagrayvid** [comment]
/hold
Want to wait a few days to give @zvonkok a chance to look over it

**Comment by zvonkok** [comment]
This looks good to me as a first shot. For the 4.9 timeframe we wanna extend this section when we introduces SRO, now it make  no sense. 

Nice work @dagrayvid and thanks @ahardin-rh for the guidance and support.

**Pull request merged**
